src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -306,7 +306,7 @@ private static string GetResourceString(ref Utf8JsonReader json, ExceptionResour
                     message = SR.Format(SR.ExpectedEndAfterSingleJson, character);
                     break;
                 case ExceptionResource.ExpectedEndOfDigitNotFound:
-                    message = SR.Format(SR.ExpectedEndOfDigitNotFound, character);
+                    message = SR.Format(nextByte >= '0' && nextByte <= '9'? SR.InvalidLeadingZeroInNumber : SR.ExpectedEndOfDigitNotFound, character);
```

Comment:
Hmm... so interestingly, the following `IncompleteContainers` test is now failing in CI, and in that case, we clearly don't want the leading zero error message. We may have to special case the exception message only for the certain cases where it makes sense (**i.e. what you had previously**). I would double check though that the exception message makes sense for some of the different edge cases:
**Incomplete:**
"[1", "[-1", "[0.1", "[0.1e-", "[0.1e-5"
**Invalid:**
"0z, "-z", "-0z", "0.e", ".5", "-.5", "0.ez"
**Leading zeroes:**
"01", "-01", "00z", "-00z", "01e", "[01"

https://github.com/dotnet/corefx/blob/c05355b90b55a9ff436a1ba9c6c1f5a508c2f3ec/src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs#L83-L84

https://dev.azure.com/dnceng/public/_build/results?buildId=321542&view=ms.vss-test-web.build-test-results-tab&runId=9312866&resultId=101135&paneView=debug
```text

Error message
Assert.Equal() Failure\r\n ↓ (pos 0)\r\nExpected: '1' is an invalid end of a number. Expect···\r\nActual: Invalid leading zero before '1'. Path: $[···\r\n ↑ (pos 0)

Stack trace
   at System.Text.Json.Tests.JsonSerializerTests.IncompleteContainers() in /_/src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs:line 80
```

I would have expected this to fail in the same way. Does this test fail locally?
https://github.com/dotnet/corefx/blob/c05355b90b55a9ff436a1ba9c6c1f5a508c2f3ec/src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs#L77-L78

Reply:
>Does this test fail locally (on netfx)?

@ahsonkhan isn't this the netfx repro?
https://dev.azure.com/dnceng/public/_build/results?buildId=321542&view=ms.vss-test-web.build-test-results-tab&runId=9312642&resultId=100913&paneView=debug

Reply:
That failure is the assert on line 80, what about the one on line 78?

>  at System.Text.Json.Tests.JsonSerializerTests.IncompleteContainers() in /_/src/System.Text.Json/tests/NewtonsoftTests/**JsonSerializerTests.cs:line 80**

Reply:
Without the fix, it does fail on line 78 on my local. Also fails on the Assert of line 84.

Here's what I did just to make sure I got the right result:
`dotnet.exe msbuild /t:RebuildAndTest ../src/System.Text.Json/tests/System.Text.Json.Tests.csproj /p:XUnitMethodName=System.Text.Json.Tests.JsonSerializerTests.IncompleteContainers /p:TargetGroup=netfx`

Reply:
> Without the fix, it does fail on line 78 on my local. Also fails on the Assert of line 84.

I see. That makes sense. It's strange that the failure reported in CI for netfx is line 80 and not the earlier assert.

Reply:
> I would double check though that the exception message makes sense for some of the different edge cases

@Jozkee - do all the exception messages for those cases make sense to you?